### DISCUSSION TITLE
Gesture interaction with model

### DISCUSSION THREAD

**Discussion by Ncit** [sceneview/sceneform-android#86] [Q&A]
Hello there, 
searching for any clue on how to zoom in/out, rotate the model for scene view-background sample. Can someone help?

**Comment by ThomasGorisse** [comment]
Use a TransformableNode like here: [samples/gltf/src/main/java/com/google/ar/sceneform/samples/gltf/MainActivity.java](samples/gltf/src/main/java/com/google/ar/sceneform/samples/gltf/MainActivity.java)

**Reply by Ncit** [reply]
Thanks for the answer. But this example requires to use of ARfragment, but I need just to show 3d model with interaction without AR. addOnPeekTouchListener for scene lack of functions(for example createAnchor) for using TransformableNode.

**Reply by Ncit** [reply]
Is it possible to disable the camera on ARfragment, so I can handle gesture interaction with the model without ar?

**Reply by ThomasGorisse** [reply]
In that case, you must use the SceneView directly like here: [samples/sceneview-background](https://github.com/ThomasGorisse/sceneform-android-sdk/tree/master/samples/sceneview-background)
but you wont actually be able to handle gestures.

If it's a pure 3D rendering you are looking for, you should definitely move to [Filament ModelViewer](https://github.com/google/filament/blob/main/android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt) without using Sceneform.

**Reply by Ncit** [reply]
I managed to zoom in/out via scaleGestureDetector. But stuck with rotation, I cannot rotate 360 degrees.
latestRotation = Quaternion(Vector3(totalyPosition, 0f, 0f), 1f)
node.localRotation = latestRotation

**Reply by Ncit** [reply]
Thanks for the response with the filament example. But I did not manage to start example with 3dmodel interaction

**Reply by ThomasGorisse** [reply]
See Filament Model viewer in filament/android/gltfio
